openmcyclus/DepleteReactor.py

```diff
@@ -8,8 +11,523 @@ class DepleteReactor(Facility):
+        uilabel="Minimum fresh fuel inventory",
+        units = "assemblies"
+    )
+
+    n_assem_spent = ts.Int(
+        doc = "Number of spent fuel assemblies that can be stored "\
+              "on-site before reactor operation stalls",
+        default = 10000000,
+        uilabel="Maximum spent fuel inventory",
+        units = "assemblies"
+    )
+
+    power_cap = ts.Double(
+        doc = "Maximum amount of power (MWe) produced",
+        tooltip = "Maximum amount of power (MWe) produced",
+        uilabel = "power_cap",
+        units = "MW",
+        default=0
+    )
+
+    decom_transmute_all = ts.String(
+        doc = "If true, the archetype transmutes all assemblies "\
+              "upon decommisioning. If false, the archetype only "\
+              "transmutes half.",
+        default = 0
+    )
+   
+    fresh_fuel = ts.ResBufMaterialInv()
+    core = ts.ResBufMaterialInv()
+    spent_fuel = ts.ResBufMaterialInv()
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fresh_fuel_entry_times = []
+        self.core_entry_times = []
+        self.fresh_fuel.capacity = self.assem_size*self.n_assem_fresh
+        self.core.capacity = self.assem_size*self.n_assem_core
+        self.spent_fuel.capacity = self.assem_size*self.n_assem_spent
+        self.cycle_step = 0
+        self.power_name = "power"
+        self.discharged = False
+        self.resource_indexes = {}
+
     def tick(self):
-        print("Hello ")
+        '''
+        Logic to implement at the tick phase of each 
+        time step. 
+        
+        If the prototype is retired, then that is recorded, 
+        fuel is transmuted, and the prototype is decommissioned. 
+
+        If it's the end of a cycle, that is recorded. If it is 
+        after a cycle ends, and fuel has not been discharged, 
+        then the fuel is discharged. If it's after a cycle ends, then 
+        fuel is loaded
+        '''
+        if self.retired():
+        #    self.record("RETIRED", "")
+            if self.context.time == self.exit_time + 1:
+                if self.decom_transmute_all == 1:
+                    self.transmute(math.ceil(self.n_assem_core))
+                else:
+                    self.transmute(math.ceil(self.n_assem_core/2))
+
+            while self.core.count > 0:
+                if self.discharge() == 0:
+                    break
+
+            while (self.fresh_fuel > 0) and (self.spent_fuel.space >= self.assem_size):
+                self.spent_fuel.push(self.fresh_fuel.pop())
+            if self.check_decommission_condition():
+                self.decommission()
+        if self.cycle_step == self.cycle_time:        
+            self.transmute(math.ceil(self.n_assem_batch))
 
+        if (self.cycle_step >= self.cycle_time) and (self.discharged == False):
+            self.discharged = self.discharge()
+
+        if self.cycle_step > self.cycle_time:
+            self.load()
```

Comment:
This could be an `if-elif-else` block

Reply:
I prefer them as separate if loops because even though the variables you are looking at for each loop are the same, the logic implemented for each loop makes more sense as different if loops instead of an if-elif loop. So comparing the actions instead of the triggering events.